### PR TITLE
Ascii Draw Method Extensions

### LINKED ISSUE #2366: Refactor `asciiDraw` routines as extension methods on `Tile`
It is my opinion that `asciiDraw`,  `asciiDrawDouble`, `createAsciiTileString`, `asciiDrawRange` shouldn't be polluting the `Tile.scala`. They seem more appropriately delivered as extension methods.  [Here](https://github.com/locationtech/geotrellis/pull/2362/commits/232adfb1e0325cdb58b7d7d329640333e6049f39) is an example of one approach to doing this.

### PR DESCRIPTION
**Pull request by jbouffard**
## Overview

This PR moves the `asciiDraw`, `asciiDrawDouble`, `asciiDrawRange`, and `createAsciiTileString` from `Tile` into their own extension methods.

### Checklist

- [x] `docs/CHANGELOG.rst` updated, if necessary
- [x] `docs` guides update, if necessary
- [x] New user API has useful Scaladoc strings

### Notes

Much of the logic from this PR was taken from #2362 

Closes #2366

### CODE REVIEW

File: raster/src/main/scala/geotrellis/raster/render/ascii/NumericEncoder.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import geotrellis.raster.{Tile, isNoData}
+
+import spire.syntax.cfor._
+
+import scala.collection.mutable.ArrayBuffer
+import scala.math.BigDecimal
+
+import java.util.Locale
+
+
```
Comment (pomadchin) [inline raster/src/main/scala/geotrellis/raster/render/ascii/NumericEncoder.scala]: There are no headers, mb it makes sense to do it in a separate PR before 2.0 release, after all issues would be closed.

**Comment by pomadchin** [comment]
Looks good, a bit confused that it's @metasim idea, mb he can have some strong opinion about htese methods?

**Comment by metasim** [comment]
LGTM. If i weren't afraid of breaking changes, I'd also change all the `asciiDraw` names to `stringMatrix` or `tableView` or `renderTable` or something else because its not really `Draw`-ing, and the `ascii` bit is confusing to me. 

The overall goal of the original PR was a) apply the principle of separation of concerns, and b) to normalize the notion of rendering a `Tile` into multiple forms: PNG, JPG, TIFF, ASCII-art, String-ified numbers, etc. I'd argue a `renderCSV` is more useful than `asciiDraw`.

**Comment by metasim** [comment]
PS: @jbouffard Thanks for resurrecting the work. I hated seeing it languish, but didn't have time to readdress it.

**Comment by jbouffard** [comment]
@metasim you're welcome! I'm glad that I could bring back your good work into GeoTrellis.

File: raster/src/main/scala/geotrellis/raster/render/ascii/NumericEncoder.scala (hunk 1 above)
Comment (echeipesh) [reply raster/src/main/scala/geotrellis/raster/render/ascii/NumericEncoder.scala]: 👍
